kernel/kselftest.py

```diff
@@ -133,19 +133,27 @@ def setUp(self):
                 self.buldir = "/usr/src/linux"
 
         self.sourcedir = os.path.join(self.buldir, self.testdir)
-        ksm_test_dir = self.sourcedir + "/vm/ksm_tests"
-        if not os.path.isfile(ksm_test_dir):
-            if build.make(self.sourcedir):
-                self.fail("Compilation failed, Please check the build logs")
+        if self.comp:
+            build_str = '-C %s' % self.comp
+        if build.make(self.sourcedir, extra_args='%s' % build_str):
+            self.fail("Compilation failed, Please check the build logs !!")
 
     def test(self):
         """
         Execute the kernel selftest
         """
         self.error = False
         kself_args = self.params.get("kself_args", default='')
-        build.make(self.sourcedir,
-                   extra_args='%s %s run_tests' % (kself_args, self.comp))
+        if self.subtest == "ksm_tests":
```

Comment:
can we target this to part of the kernel selftest case enhancement as like  user can pass component to selftest as parm so it will help us on executing other components, even we can also avoid this hard codding

Reply:
Yeah that is an option we should definitely pursue independently. But whether it gets implemented or not we will need to have such a check to maintain backward compatibility. All existing and past distros would need this check.

One other option to avoid this check is to separate out KSM part from selftest and write an independent test for memory sub component.

Reply:
adding a new test case can be an option, but we need to have a repetition of the code base, what I am thinking we can this commit, and have more focus on around keslf test case which help us on longer goal (enabling other test cases) , it sounds ok to you(@sacsant ) please let me know

Reply:
Sounds like a plan.